src/main/java/org/truffleruby/extra/ffi/Pointer.java

```diff
@@ -90,21 +90,27 @@ public void writeBytes(long offset, long size, byte value) {
         UNSAFE.setMemory(address + offset, size, value);
     }
 
-    public void writeBytes(long offset, byte[] bytes, int index, int length) {
-        for (int n = 0; n < length; n++) {
-            writeByte(offset + n, bytes[index + n]);
-        }
+    public void writeBytes(long offset, byte[] buffer, int bufferPos, int length) {
+        assert address + offset != 0 || length == 0;
+        assert buffer != null;
+        assert bufferPos >= 0;
+        assert length >= 0;
+
+        UNSAFE.copyMemory(buffer, Unsafe.ARRAY_BYTE_BASE_OFFSET + bufferPos, null, address + offset, length);
```

Comment:
Is it faster than the loop? By how much?

Reply:
Not massively. For the revcomp benchmark, it shaves ~7% off the time for `readlines`. But it's much more pronounced in the native image where there's a higher penalty for function calls.

Reply:
I should note I haven't extensively benchmarked. I've really only looked at the one from #792. I believe `Unsafe` has optimized copy operations for aligned reads & writes.